compaction/compaction.cc

```diff
@@ -729,6 +731,11 @@ class compaction {
 
         _ms_metadata.min_timestamp = timestamp_tracker.min();
         _ms_metadata.max_timestamp = timestamp_tracker.max();
+
+        size_t max_overlapping = std::ranges::max(token_map | boost::adaptors::transformed([] (auto& interval) {
+            return interval.second.size();
+        }));
```

Comment:
std::ranges::max supports projection, so you should be able to write

```c++
std::ranges::max(token_map, std::less<>(), [] (const interval_type& interval) { return interval.second.size(); });
```

Reply:
will change, thanks.